### PR TITLE
Encapsulate FSDPA in GaudiLlamaAttention

### PR DESCRIPTION
**Pull request by dudilester**
* Done to allow quantization using HQT

* Added use_flash_attention and flash_attention_recompute to run_lm_eval

### CODE REVIEW

**Review by ulivne — CHANGES_REQUESTED** [review]

File: optimum/habana/transformers/models/llama/modeling_llama.py [hunk 1]
```diff
@@ -309,18 +320,20 @@ def pre_attn_forward(
             if q_len == 1:
                 # next token
                 with ht.sdp_kernel(enable_recompute=False):
-                    attn_output = FusedSDPA.apply(
+                    attn_output = self.fused_scaled_dot_product_attention.forward(
```
[the remark this answers is not in the document]
Comment (dudilester) [reply optimum/habana/transformers/models/llama/modeling_llama.py]: Done

File: optimum/habana/transformers/models/llama/modeling_llama.py [hunk 2]
```diff
@@ -309,18 +320,20 @@ def pre_attn_forward(
...
                         query_states, key_states, value_states, attention_mask, 0.0, False, None
                     )
             else:
                 # first token
                 if flash_attention_causal_mask:
                     # causal masking on first token requires inputs to be of the same lenght
                     with ht.sdp_kernel(enable_recompute=flash_attention_recompute):
-                        attn_output = FusedSDPA.apply(query_states, key_states, value_states, None, 0.0, True, None)
+                        attn_output = self.fused_scaled_dot_product_attention.forward(
```
[the remark this answers is not in the document]
Comment (dudilester) [reply optimum/habana/transformers/models/llama/modeling_llama.py]: These lines are under the condition that FusedSDPA exists, same condition that cause the instantiation of the module in line 177

**Review by MrGeva — APPROVED** [review]

**Comment by astachowiczhabana** [comment]
https://github.com/huggingface/optimum-habana/pull/972

**Comment by dudilester** [comment]
upstream URL
https://github.com/huggingface/optimum-habana/pull/976

**Pull request merged**